### PR TITLE
add meson instructions to usage docs

### PR DESCRIPTION
**Pull request by fennewald**
Adds a usage section for meson to the docs.

### CODE REVIEW

File: doc/usage.rst [hunk 1]
```diff
@@ -134,6 +134,44 @@ For ``build2`` newcomers or to get more details and use cases, you can read the
 ``build2``
 `toolchain introduction <https://build2.org/build2-toolchain/doc/build2-toolchain-intro.xhtml>`_.
 
+Usage with meson
```
Comment (vitaut) [inline doc/usage.rst:137]: Meson should be capitalized.

File: doc/usage.rst [hunk 2]
```diff
@@ -134,6 +134,44 @@ For ``build2`` newcomers or to get more details and use cases, you can read the
...
+================
+
+`Meson's wrapdb <https://mesonbuild.com/Wrapdb-projects.html>` includes a ``fmt``
```
Comment (vitaut) [inline doc/usage.rst:140]: Throughout the section:

wrapdb -> WrapDB

File: doc/usage.rst [hunk 3]
```diff
@@ -134,6 +134,44 @@ For ``build2`` newcomers or to get more details and use cases, you can read the
...
+    fmt = subproject('fmt')
+    fmt_dep = fmt.get_variable('fmt_dep')
+
+- Include the new dependency object to link with fmt::
+
+    my_build_target = executable('name', 'src/main.cc', dependencies: [fmt_dep])
+
+**Options:**
+
+If desired, ``fmt`` may be built as a static library, or as a headers-only
```
Comment (vitaut) [inline doc/usage.rst:162]: Here an below:

headers-only -> header-only

**Review by vitaut — CHANGES_REQUESTED** [review]
Thanks for the PR. Mostly looks good, just a few minor comments inline.

**Comment by fennewald** [comment]
Changes applied and squashed into a single commit. Let me know if you'd like any other changes.

Thank you for the work you do maintaining this valuable library!

**Review by vitaut — APPROVED** [review]

**Pull request merged**